### PR TITLE
Add cache config form to question sidebar

### LINKED ISSUE #42731: Add caching policy form to question info sidebar

### PR DESCRIPTION
**Pull request by rafpaf**
This PR adds a configuration form to the question sidebar. To launch it, visit a question, open the info sidebar (the "i" icon), and you'll see **Caching policy**, plus the name of the current caching policy for that question. Click the policy to open the form.

Closes https://github.com/metabase/metabase/issues/42731

### CODE REVIEW

**Comment by rafpaf** [comment]
* **#42729** <a href="https://app.graphite.dev/github/pr/metabase/metabase/42729?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rafpaf and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: frontend/src/metabase/common/components/performance/SidebarCacheSection.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import type { Dispatch, SetStateAction } from "react";
```
Comment (rafpaf) [inline frontend/src/metabase/common/components/performance/SidebarCacheSection.tsx:1]: This is the form-launching component found in both the dashboard and question sidebars

**Comment by npfitz** [comment]
Only thing I see here is how we show the caching page. Rather than set up a specific sidebar, we are kinda hijacking the `<QuestionInfoSidebar/>` component. I don't think it's a problem per say, but it does go against the pattern the other sidebars use. If we could get a thumbs up from the team that owns that code, then I think this is good to merge.

**Comment by kulyk** [comment]
> Only thing I see here is how we show the caching page. Rather than set up a specific sidebar, we are kinda hijacking the `<QuestionInfoSidebar/>` component. I don't think it's a problem per say, but it does go against the pattern the other sidebars use. If we could get a thumbs up from the team that owns that code, then I think this is good to merge.

LGTM! Don't think there are massive pros/cons between the two approaches; even if one appears to be less convenient one day, it doesn't seem hard to tweak 👍

**Comment by rafpaf** [comment]
Thank you @kulyk 🙏🙏

**Review by npfitz — APPROVED** [review]

**Pull request merged**